### PR TITLE
Move article back to WPF area

### LINKED ISSUE #16291: Move xaml-services-types-migrated article to migration guide.
https://docs.microsoft.com/en-us/dotnet/framework/xaml-services/types-migrated-from-wpf-to-system-xaml should really be in the migration guide. It's talking about WPF changes from 3.5 onward. This should be removed from the xaml-services library.

### PR DESCRIPTION
**Pull request by Thraka**
## Summary

- Moved file about xaml.
- Updated redirects.
- Added to TOC in advanced.
- Edited TOC index-type file for migration.

Fixes #16291

### CODE REVIEW

File: docs/framework/wpf/advanced/toc.yml [hunk 1]
```diff
@@ -481,6 +481,8 @@
   - name: Migration and Interoperability
     href: migration-and-interoperability.md
     items:
+    - name: Migrating from WPF 3.X to WPF 4.5
```
Comment (gewarren) [inline docs/framework/wpf/advanced/toc.yml]:
```suggestion
    - name: Migrate from WPF 3.X to WPF 4.5
```

File: docs/framework/wpf/advanced/types-migrated-from-wpf-to-system.md [hunk 2]
```diff
@@ -7,9 +7,10 @@ helpviewer_keywords:
...
+In .NET Framework 3.5 and .NET Framework 3.0, both Windows Presentation Foundation (WPF) and Windows Workflow Foundation included a XAML language implementation. Many of the public types that provided extensibility for the WPF XAML implementation existed in the WindowsBase, PresentationCore, and PresentationFramework assemblies. Likewise, public types that provided extensibility for Windows Workflow Foundation XAML existed in the System.Workflow.ComponentModel assembly. In the .NET Framework 4, some of the XAML-related types are migrated to the System.Xaml assembly. A common .NET Framework implementation of XAML language services enables many XAML extensibility scenarios that were originally defined by a specific framework's XAML implementation but are now part of overall .NET Framework 4 XAML language support. This topic lists the types that are migrated and discusses issues related to the migration.
```
Comment (gewarren) [inline docs/framework/wpf/advanced/types-migrated-from-wpf-to-system.md]:
```suggestion
In .NET Framework 3.5 and .NET Framework 3.0, both Windows Presentation Foundation (WPF) and Windows Workflow Foundation included a XAML language implementation. Many of the public types that provided extensibility for the WPF XAML implementation existed in the WindowsBase, PresentationCore, and PresentationFramework assemblies. Likewise, public types that provided extensibility for Windows Workflow Foundation XAML existed in the System.Workflow.ComponentModel assembly. In the .NET Framework 4, some of the XAML-related types were migrated to the System.Xaml assembly. A common .NET Framework implementation of XAML language services enables many XAML extensibility scenarios that were originally defined by a specific framework's XAML implementation but are now part of overall .NET Framework 4 XAML language support. This article lists the types that were migrated and discusses issues related to the migration.
```

are migrated -> were migrated

File: docs/framework/wpf/advanced/toc.yml (hunk 1 above)
Comment (gewarren) [reply docs/framework/wpf/advanced/toc.yml]: Do you also need to remove the article from the desktop-wpf TOC?

**Review by gewarren — APPROVED** [review]
Just a couple suggestions.

File: docs/framework/wpf/advanced/toc.yml (hunk 1 above)
Comment (Thraka) [reply docs/framework/wpf/advanced/toc.yml]: oh yeah, duh! ha!

**Pull request merged**